edit-post/hooks/autocompleters/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies.
+ */
+import { find } from 'lodash';
+
+/**
+ * WordPress dependencies.
+ */
+import { addFilter } from '@wordpress/hooks';
+import { select, dispatch } from '@wordpress/data';
+import { createBlock } from '@wordpress/blocks';
+
+// Export for unit test
+export function getBlockOptions( query ) {
+	const editor = select( 'core/editor' );
+	const isInitialQuery = ! query;
+
+	return isInitialQuery ?
+		// Before we have a query, offer frecent blocks as a sensible default.
+		editor.getFrecentInserterItems() :
+		editor.getInserterItems();
+}
+
+// Export for unit test
+export function getBlockCompletion( inserterItem ) {
+	const { name, initialAttributes } = inserterItem;
+	return {
+		action: 'replace',
+		value: createBlock( name, initialAttributes ),
+	};
+}
+
+// Export for unit test
+export function addReusableBlocksCompletion( completers ) {
+	const blocksCompleter = find( completers, c => 'blocks' === c.name );
```

Comment:
Can it be expressed with destructuring to avoid using a meaningful name `c`?
```js
const blocksCompleter = find( completers, ( { name } ) => 'blocks' === name );
```